### PR TITLE
Make query-based sync default sync mode

### LINKED ISSUE #1793: Make Query-based Sync the default mode for syncing data
All bindings need to adopt Query-based synchronization (formerly named Partial Sync) as the default mode of synching. This is most likely a breaking API change which requires a major API bump.

Java merged their PR here: realm/realm-java#5893

In particular the setting on SyncConfiguration was renamed from partialRealm() to fullSynchronization().

- [ ] Change API
- [ ] Update documentation to use Query-based Sync instead of Partial Sync

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes #1793 

This is a breaking change, so version has been bumped to 3.0.0
All public methods and docs should now use the term query-based sync instead of partial sync.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+### Compatibility
+* Sync protocol: 24
+* Server-side history format: 4
+* File format: 7
+* Realm Object Server: 3.0.0 or later
+
+### Breaking changes
+Partial synchronization has been renamed to Query-based synchronization and the API is now considered stable and have left beta.
+
+* [Sync] Query-based sync is now the default mode for synced Realms. Current fully synced Realms should add `full_synchronization: true` to their `sync` configuration to maintain their current behaviour.
```
Comment (bmunkholm) [inline CHANGELOG.md]: What happens if they don't?

**Comment by cmelchior** [comment]
Updated. I'll probably wait to merge this until we are closer to release. Otherwise, we are prevented from making any more minor versions in the meantime.

**Comment by bmunkholm** [comment]
@cmelchior As we discussed, we need to find a method of not breaking the API at runtime but compile time.

**Comment by cmelchior** [comment]
We decided to move the API to `SyncUser` instead. Closing this.

**Pull request closed without merging**